### PR TITLE
Add tests bad type_support implementation

### PR DESCRIPTION
**Pull request by Lobotuerk**
This tests should add 10 lines of coverage to `rmw_fastrtps_cpp`
Signed-off-by: lobotuerk <jtlorente@ekumenlabs.com>

### CODE REVIEW

File: test_rmw_implementation/test/test_subscription.cpp [hunk 1]
```diff
@@ -150,6 +150,14 @@ TEST_F(CLASSNAME(TestSubscription, RMW_IMPLEMENTATION), create_with_bad_argument
   EXPECT_EQ(nullptr, sub);
   rmw_reset_error();
 
+  rosidl_message_type_support_t * non_implementation_ts = const_cast<rosidl_message_type_support_t *>(ts);
```
Comment (hidmic) [inline test_rmw_implementation/test/test_subscription.cpp:153]: @Lobotuerk nit: 
```suggestion
  rosidl_message_type_support_t * non_const_ts = const_cast<rosidl_message_type_support_t *>(ts);
```
Same elsewhere.

File: test_rmw_implementation/test/test_subscription.cpp [hunk 2]
```diff
@@ -150,6 +150,14 @@ TEST_F(CLASSNAME(TestSubscription, RMW_IMPLEMENTATION), create_with_bad_argument
...
+  const char * old_identifier = non_implementation_ts->typesupport_identifier;
+  non_implementation_ts->typesupport_identifier = "not-an-implementation-identifier";
```
Comment (hidmic) [inline test_rmw_implementation/test/test_subscription.cpp:155]: @Lobotuerk nit: 
```suggestion
  non_implementation_ts->typesupport_identifier = "not-a-typesupport-identifier";
```
Same elsewhere.

File: test_rmw_implementation/test/test_subscription.cpp [hunk 3]
```diff
@@ -150,6 +150,14 @@ TEST_F(CLASSNAME(TestSubscription, RMW_IMPLEMENTATION), create_with_bad_argument
   EXPECT_EQ(nullptr, sub);
   rmw_reset_error();
 
+  rosidl_message_type_support_t * non_implementation_ts = const_cast<rosidl_message_type_support_t *>(ts);
+  const char * old_identifier = non_implementation_ts->typesupport_identifier;
```
Comment (hidmic) [inline test_rmw_implementation/test/test_subscription.cpp:154]: @Lobotuerk nit: 
```suggestion
  const char * typesupport_identifier = non_implementation_ts->typesupport_identifier;
```
Same elsewhere.

**Review by hidmic — APPROVED** [review]
LGTM pending green CI

Thanks for the extra lines @Lobotuerk !

File: test_rmw_implementation/test/test_subscription.cpp (hunk 3 above)
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_subscription.cpp]: Updated on f1fc248

File: test_rmw_implementation/test/test_subscription.cpp (hunk 2 above)
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_subscription.cpp]: Updated on f1fc248

File: test_rmw_implementation/test/test_subscription.cpp (hunk 3 above)
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_subscription.cpp]: Updated on 345697b

**Comment by Lobotuerk** [comment]
CI up `rcl` and `test_rmw_implementation`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12572/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7540/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10283/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12497/)

File: test_rmw_implementation/test/test_serialize_deserialize.cpp [hunk 4]
```diff
@@ -57,6 +57,18 @@ TEST_F(CLASSNAME(TestSerializeDeserialize, RMW_IMPLEMENTATION), serialize_with_b
...
+    const_cast<rosidl_message_type_support_t *>(ts);
+  const char * typesupport_identifier = non_const_ts->typesupport_identifier;
+  non_const_ts->typesupport_identifier = "not-a-typesupport-identifier";
+  rcutils_allocator_t default_allocator = rcutils_get_default_allocator();
+  ASSERT_EQ(
+    RMW_RET_OK, rmw_serialized_message_init(
+      &serialized_message, 0lu, &default_allocator)) << rmw_get_error_string().str;
+  EXPECT_NE(RMW_RET_OK, rmw_serialize(&input_message, non_const_ts, &serialized_message));
+  rmw_reset_error();
+  non_const_ts->typesupport_identifier = typesupport_identifier;
```
Comment (hidmic) [inline test_rmw_implementation/test/test_serialize_deserialize.cpp:70]: @Lobotuerk if the `ASSERT_EQ` above fails, the typesupport struct will be left in a broken state.

Comment (hidmic) [reply test_rmw_implementation/test/test_serialize_deserialize.cpp:76]: See https://github.com/ros2/rmw_implementation/pull/152/commits/af19cb4dce63ec866d9597589c694277feba8a3f.

**Comment by hidmic** [comment]
Another CI run:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12574/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7542/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10285/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12499/)

**Review by Blast545 — APPROVED** [review]
LGTM

**Comment by hidmic** [comment]
CI again after https://github.com/ros2/rmw_fastrtps/pull/458:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12590/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7557/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10302/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12515/)

**Comment by hidmic** [comment]
Alright, all green ! Thanks @Lobotuerk !

**Pull request merged**